parsl/dataflow/dflow.py

```diff
@@ -1181,9 +1177,9 @@ def add_executors(self, executors: Sequence[ParslExecutor]) -> None:
         for executor in executors:
             executor.run_id = self.run_id
             executor.run_dir = self.run_dir
-            executor.hub_address = self.hub_address
-            executor.hub_zmq_port = self.hub_zmq_port
             if self.monitoring:
+                executor.hub_address = self.monitoring.hub_address
+                executor.hub_zmq_port = self.monitoring.hub_zmq_port
                 executor.monitoring_radio = self.monitoring.radio
```

Comment:
I'm wondering if it would be better to simply attach the monitor to the executor, rather than copying these details over piecemeal.  I haven't thought that through though, I just don't like the smell of multiple copies of the same data lying around.

Reply:
the whole model of populating state post-construction ad-hoc into various objects makes me generally uncomfortable in Parsl - way longer term I'd love some stronger notion of "existing in a context" that doesn't look like this. (absolutely not suggesting this as an approach but compare how pytest uses fixtures)

that's not for this PR

Reply:
although monitoring specific, this is part of the monitoring radio plugin prototype which I hope to reduce the context that needs to be passed around to be "here's the object you use to send your messages, it encapsulates everything needed for that".